realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -361,7 +361,7 @@ private static void initializeRealm(Realm realm) {
             }
         } catch (Exception e) {
             commitChanges = false;
-            throw e;
+            throw new RuntimeException(e);
```

Comment:
This was an attempt to get to Java 6 compatibility, as requested above.  As you know, Java 6 does not support inferring the Exception type.  Removed.